<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Exercise 1 (Learning to interrupt)  

In [15]:
while True:
    print('*', end='')
    
#interrupt kernal via 'kernal' > 'interrupt kernal' or press shortcut 'i i' 
#message "KeyboardInterrupt: " is shown as output. 

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

## Exercise 2 (Resetting & Restarting the Kernal)  

In [16]:
print = 'PRINT'

In [6]:
print('Hello World')

Hello World


## Exercise 3 (Timing Stuff)

In [11]:
%%timeit
squares = [i**2 for i in range(10)]

252 ns ± 1.3 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [14]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

285 ns ± 2.56 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The first function processes output faster than the second function (252ns vs 285ns) after running a total of 7,000,000 loops. 
1 run is 1,000,000 loops. A total of 7 runs was runned, leading to a total of 7,000,000 loops runned. 
The output timing (252+/-sd and 282ns+/-sd) is the mean+SD of the all 7,000,000 loops runned. 